app/models/course/experience_points_disbursement.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+class Course::ExperiencePointsDisbursement
+  include ActiveModel::Model
+  include ActiveModel::Validations
+
+  attr_accessor :reason, :course_users
+  attr_reader :experience_points_records
+
+  validates :reason, presence: true
+
+  # Builds new experience points records for each +CourseUser+ in the given list of course users.
+  # Also stores the list of course users for later reference.
+  #
+  # @param [Array<CourseUser>] course_users Course users to build records for
+  # @return [Array<Course::ExperiencePointsRecord>] The newly built experience points records
+  def build_points_records_for(course_users)
+    @course_users = course_users
+    @course_users.map { |course_user| course_user.experience_points_records.build }
+  end
+
+  # Processes course user attributes and builds experience points records for course users that
+  # should be awarded points.
+  #
+  # @param [Hash] attributes Course user attributes
+  def course_users_attributes=(attributes)
+    @experience_points_records = attributes.values.map do |course_user_hash|
+      points_awarded = course_user_hash['experience_points_record']['points_awarded']
+      if points_awarded.empty? || points_awarded == '0'
+        nil
+      else
+        Course::ExperiencePointsRecord.new(course_user_id: course_user_hash['id'],
```

Comment:
this should go find the appropriate course_user by ID then `call experience_points_records.build`